server/app/uk/gov/ons/addressIndex/server/modules/validation/RandomControllerValidation.scala

```diff
@@ -90,4 +90,28 @@ class RandomControllerValidation @Inject()(implicit conf: ConfigModule, versionP
     }
   }
 
+   def validateBoosts(eboost: Option[String],nboost: Option[String],sboost: Option[String],wboost: Option[String],queryValues: QueryValues): Option[Future[Result]] = {
+    val eboostVal = {if (eboost.getOrElse("1.0").isEmpty) "1.0" else eboost.getOrElse("1.0")}
+    val nboostVal = {if (nboost.getOrElse("1.0").isEmpty) "1.0" else nboost.getOrElse("1.0")}
+    val sboostVal = {if (sboost.getOrElse("1.0").isEmpty) "1.0" else sboost.getOrElse("1.0")}
+    val wboostVal = {if (wboost.getOrElse("1.0").isEmpty) "1.0" else wboost.getOrElse("1.0")}
```

Comment:
....

Reply:
The more verbose code is required here because of the case where the query string parameter is supplied but not given a value - this comes through as an empty string, not a None
e.g. http://localhost:9001/addresses/partial?input=7%20gate%20reach&eboost=
which gives an error with just val eboostVal = eboost.getOrElse("1.0") (I have tested it)

Reply:
Hmmm. Sounds like an issue further up then. Since we're dealing with `Options` should the blank value come through as a `None` not an empty `String`.

Reply:
Declaring all the query string params as Option[String] means we can validate them better and give more consistent error messages, at the cost of doing a bit more work to process them. The alternative is to type the parameters "natively" as ints, doubles, booleans etc. and used Play's rather basic built-in validation.

Reply:
Ok this still smells funny to me. It looks like you can still have `val eboostVal = eboost.getOrElse("1.0")` as further down you have `val eboostDouble = Try(eboostVal.toDouble).toOption.getOrElse(1.0D)` - this will always return a value even if there's nothing in `eboostVal`. So this will work with `eboost = None, Some(""), Some("1.3")` etc.

Reply:
The eboostVal values go into the validation method, so the user will be told if an invalid value has been supplied. The doubles need to exist whatever for the response fields.

Reply:
At the moment it's the `eboost = Some(eboostDouble),` going to the validation method.

Reply:
No, the validateBoosts function takes the string parameters and recreates the doubles with a value of 99 for a non-numeric value.

val eboostDouble = Try(eboostVal.toDouble).toOption.getOrElse(99D)

    else if (eboostDouble > 10 || nboostDouble > 10 || sboostDouble > 10 || wboostDouble > 10 || eboostDouble < 0 || nboostDouble < 0 || sboostDouble < 0 || wboostDouble < 0) {
      logger.systemLog(badRequestMessage = CountryBoostsInvalidError.message)
      Some(futureJsonBadRequest(PartialCountryBoostsInvalid(queryValues))) 

This could be extended to report a different message for a non-numeric value rather than an out of range number, if we so wished, for now there is a single message. We could also test the values individually - debatable if this is more or less user-friendly.

Reply:
Ok agreed that that is the case in the Validation classes. However, in the Controllers: 
```  
    val eboostVal = {if (eboost.getOrElse("1.0").isEmpty) "1.0" else eboost.getOrElse("1.0")}
    val nboostVal = {if (nboost.getOrElse("1.0").isEmpty) "1.0" else nboost.getOrElse("1.0")}
    val sboostVal = {if (sboost.getOrElse("1.0").isEmpty) "1.0" else sboost.getOrElse("1.0")}
    val wboostVal = {if (wboost.getOrElse("1.0").isEmpty) "1.0" else wboost.getOrElse("1.0")}

    val eboostDouble = Try(eboostVal.toDouble).toOption.getOrElse(1.0D)
    val nboostDouble = Try(nboostVal.toDouble).toOption.getOrElse(1.0D)
    val sboostDouble = Try(sboostVal.toDouble).toOption.getOrElse(1.0D)
    val wboostDouble = Try(wboostVal.toDouble).toOption.getOrElse(1.0D)`
```
The `...Val`'s aren't used anywhere else apart from this part of code.

Reply:
It is true that the Vals are not used in the controller other than as an intermediate step to the doubles (the validator takes the params direct), but I can see that this is because they haven't been added to the logging which they probably should be.

Reply:
I will push a change with the logging updated rather than removing the vals

Reply:
If they are logged in the validator do we need to log them in the controller as well?

Reply:
This is the routine logging for Splunk, we log the parameters used for each request and these can be queried / used in dashboards and reports.

Reply:
Note that this is also the logging we were going to change to suit Stackdriver

Reply:
Ok lets end this convo and get it merged.